### DISCUSSION TITLE
Using more complex geometries

### DISCUSSION THREAD

**Discussion by Mitko88** [pylbm/pylbm#146] [General]
Will the algorithms work if more complex geometries are supplied (e.g., to supply an array of binary values representing obstacles). I want to simulate the propagation of gasses in real-world scenarios in 2D and 3D, and I would need to use more complex geometries.

**Comment by bgraille** [comment]
Hi @Mitko88,
thank you for your question. For the present stable version of pylbm, the geometry can be built with the proposed geometric shapes. We are currently adding the functionality to built the geometry from STL files: it is functional but we would like to add some additional features (moving, rotating, scaling the interface).
To answer precisely, it is not possible at this time to describe the geometry only with the inner/outer points (binary values) as the Lattice Boltzmann method need also the distance to the obstacle, the normal vector at the interface... But this type of information can be built as a pretreatment and we could try to add this feature. You think about LAS file ?

**Reply by Mitko88** [reply]
Hi @bgraille,
Thank you for the explanation. LAS is fine, it is used for point clouds, and you can store additional columns there. I have a lot of rasterized/voxelized models as arrays, they can be exported in many ways. I guess the distance to obstacles using distance transform and the normal vector can be calculated automatically. For me anything is fine, I can precalculate them.
